config/routes.rb

```diff
@@ -14,7 +14,11 @@
     resources :items, only: [:new, :create, :edit, :update, :destroy]
   end
 
-  resources :shops, only: [:index]
+  resources :shops, only: [:index, :show]
+
+  namespace :shop do
```

Comment:
So to make everything above work we need to slightly alter this route. What we could do instead is:

```ruby
  resources :shop, path: ":shop", as: :shop do
    ...
  end
```